File: frontend/src/types/segment.ts
export interface SegmentRules {
  has_email?: boolean
  has_phone?: boolean
  has_website?: boolean
  mx_valid_email?: boolean
  non_role_email?: boolean
  min_employee_count?: number
  max_employee_count?: number
  revenue_range_in?: string[]
  status_in?: string[]
  assigned?: boolean
  unassigned?: boolean
  tags_any?: string[]
  tags_all?: string[]
  place_types_any?: string[]
  place_types_all?: string[]
}

export interface Segment {
  id: string
  name: string
  description: string | null
  weight: number
  rules: SegmentRules
  enabled: boolean
  created_at: number
  updated_at: number
}

export interface SegmentCreate {
  name: string
  description?: string | null
  weight: number
  rules: SegmentRules
  enabled: boolean
}

export interface SegmentUpdate {
  name?: string
  description?: string | null
  weight?: number
  rules?: SegmentRules
  enabled?: boolean
}

export interface SegmentPreview {
  matches: number
  total: number
}

export interface SegmentTuning {
  segment_id: string
  name: string
  matched: number
  contacted: number
  qualified: number
  conversion_rate: number
  current_weight: number
  proposed_weight: number
  delta: number
  sufficient_data: boolean
}

export interface SegmentTuningApplied {
  applied: SegmentTuning[]
  rescored: number
}

File: backend/app/schemas/segment.py
from typing import Any

from pydantic import BaseModel, Field


class SegmentBase(BaseModel):
    name: str
    description: str | None = None
    weight: int = Field(50, ge=0, le=100)
    rules: dict[str, Any] = {}
    enabled: bool = True


class SegmentCreate(SegmentBase):
    pass


class SegmentUpdate(BaseModel):
    name: str | None = None
    description: str | None = None
    weight: int | None = Field(None, ge=0, le=100)
    rules: dict[str, Any] | None = None
    enabled: bool | None = None


class SegmentRead(SegmentBase):
    id: str
    created_at: int
    updated_at: int

    model_config = {"from_attributes": True}


class SegmentPreview(BaseModel):
    matches: int
    total: int


class SegmentTuning(BaseModel):
    """Proposed weight adjustment for a single segment based on lead outcomes."""
    segment_id: str
    name: str
    matched: int
    contacted: int
    qualified: int
    conversion_rate: float
    current_weight: int
    proposed_weight: int
    delta: int
    sufficient_data: bool


class SegmentTuningApplied(BaseModel):
    applied: list[SegmentTuning]
    rescored: int
